server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1347 @@
+xml2js = require 'xml2js'
+request = require 'request'
+async = require 'async'
+moment = require 'moment'
+cozydb = require 'cozydb'
+File = require '../models/file'
+
+parser = new xml2js.Parser()
+builder = new xml2js.Builder headless: true
+
+# TODO :
+# Move updateOrCreate in a lib
+# multi account
+# multi contract
+
+# Models
+Client = cozydb.getModel 'Client',
+    clientId: String
+    vendor: String
+    numeroAcc: String
+    address: Object
+    name: Object
+    email: String
+    telMobile: String
```

Comment:
Sure, cozy models can be really specific and contain a lot of data. My concern here is about reusability. It is kind of pointless to have too specific data if it cannot be handled properly by another app. And some data is purely vendor-side data and may not be of any interest for the end user.

Typically, storing the vendor-side identifier of the client inside the Cozy is not useful, as long as we already have a cozy-specific mean of linking items together.

Also, migrating models is a complicated task, requiring to write migrations and handle existing data carefully, so I think introducing new models should be done with extra care :)



Typical example of what I am trying to say above is the new `Contract` model. It might be interesting to have it, and it compares to Weboob `Subscription` (so, it could be reused for any connector relying on Weboob). However, while the Weboob `Subscription` has only 4 specific fields, the `Contract` here has 19 fields. I feel like some of them are vendor-specific (`clientId`, `number`), others are over specific for this particular provider (`numeroDepannage`).

What I would have done is (example for the `Contract` module):
* Define all the models as generic as possible in the cozy models (a `Contract` model as generic as possible, comparable to Weboob `Subscription`).
* Subclass this model, also in the `models` dir, with [use-case](https://github.com/cozy-labs/konnectors/pull/568/files#diff-265233cf1488889e5feae307df30e3b1R16)-specific fields added (typically, this would become an `EnergyProviderContract` or something like this).
* Discard any info too specific to this particular provider, which a general energy provider would not necessarily offer.

This is the same issue with `Bill` vs `InternetBill` vs `PhoneBill` and as far as I know the strategy now is to use `Bill` only.

I am quite new to Cozy and Konnectors, so @m4dz @nicofrand or @ZeHiro will have a better overview of the subject.